src/pages/namespace/Instances/components/Filters/index.tsx

```diff
@@ -0,0 +1,265 @@
+        // For type safety, one separate return is required below for every type
+        // so it is possible to assert filters[field]?.value is defined and TS
+        // does not merge the different possible types of filters[field]?.value
+
+        if (field === "AS") {
+          return (
+            <ButtonBar key={field}>
+              <Button variant="outline">
+                {t([`pages.instances.list.filter.field.${field}`])}
+              </Button>
+              <Popover
+                open={activeMenu === field}
+                onOpenChange={(state) => handleOpenChange(state, field)}
+              >
+                <PopoverTrigger asChild>
+                  <Button variant="outline">{filters[field]?.value}</Button>
+                </PopoverTrigger>
+                <PopoverContent align="start">
+                  {field === "AS" && (
+                    <TextInput
+                      field={field}
+                      setFilter={setFilter}
+                      clearFilter={clearFilter}
+                      value={filters[field]?.value}
+                    />
+                  )}
+                </PopoverContent>
+              </Popover>
+              <Button variant="outline" icon>
+                <X onClick={() => clearFilter(field)} />
+              </Button>
+            </ButtonBar>
+          );
+        }
+
+        if (field === "STATUS" || field === "TRIGGER") {
+          return (
+            <ButtonBar key={field}>
+              <Button variant="outline">
+                {t([`pages.instances.list.filter.field.${field}`])}
+              </Button>
+              <Popover
+                open={activeMenu === field}
+                onOpenChange={(state) => handleOpenChange(state, field)}
+              >
+                <PopoverTrigger asChild>
+                  <Button variant="outline">{filters[field]?.value}</Button>
+                </PopoverTrigger>
+                <PopoverContent align="start">
+                  {field === "STATUS" && (
+                    <Options
+                      field={field}
+                      value={filters[field]?.value}
+                      setFilter={setFilter}
+                    />
+                  )}
+                  {field === "TRIGGER" && (
+                    <Options
+                      field={field}
+                      value={filters[field]?.value}
+                      setFilter={setFilter}
+                    />
+                  )}
+                </PopoverContent>
+              </Popover>
+              <Button variant="outline" icon>
+                <X onClick={() => clearFilter(field)} />
+              </Button>
+            </ButtonBar>
+          );
+        }
+
+        if (field === "AFTER" || field == "BEFORE") {
+          const dateValue = filters[field]?.value;
+          if (!dateValue) {
+            console.error("Early return: dateValue is not defined");
+            return <></>;
```

Comment:
Could we just return `null` here? Even if the result is the same, I tend to use null over empty fragments when I can (and TypeScript doesn't complain here, I just tried it) because I find it semantically a bit more suitable to what it actually does. But it's just a suggestion, you can keep it that way, if you like.

Reply:
`null` makes a lot more sense, thank you! I was unhappy about this, but wasn't aware this is possible. Fixed in 046073f86c339fde8545b04bba646957f79ea3a0